components/registry-facade/pkg/blobserve/blobserve.go

```diff
@@ -182,70 +200,95 @@ func (reg *Server) Prepare(ctx context.Context, ref string) (err error) {
 
 func (reg *Server) blobFor(ctx context.Context, ref string, readOnly bool) (fs http.FileSystem, err error) {
 	reg.mu.RLock()
-	dgst := reg.refcache[ref]
+	info, exists := reg.refcache[ref]
 	reg.mu.RUnlock()
 
-	state := blobUnknown
-	if dgst != "" {
-		fs, state = reg.blobspace.Get(dgst)
-	}
-
-	if state == blobUnknown {
-		if readOnly {
```

Comment:
this bit is necessary to prevent download when someone access external blobserve endpoint